### PR TITLE
Feat/region provider

### PR DESCRIPTION
**Pull request by Vohizy**

### CODE REVIEW

File: coverage/coverage-final.json
Comment (Sanmandresy) [inline coverage/coverage-final.json:1]: remove from this PR.

File: coverage/lcov-report/base.css
Comment (Sanmandresy) [inline coverage/lcov-report/base.css:1]: remove from this PR.

File: coverage/lcov-report/block-navigation.js
Comment (Sanmandresy) [inline coverage/lcov-report/block-navigation.js:1]: remove from this PR.

File: coverage/lcov-report/favicon.png
Comment (Sanmandresy) [inline coverage/lcov-report/favicon.png:1]: remove from this PR.

File: src/providers/axios.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
+
+export const InstanceAxiosUrl = axios.create({
```
Comment (Sanmandresy) [inline src/providers/axios.ts:3]: rename to api.

File: src/providers/axios.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  baseURL: "BASE_URL_API",
```
Comment (Sanmandresy) [inline src/providers/axios.ts:4]: why are you putting a mock url ?

File: src/types/user.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export type User = {
+  username: string;
+  password: string;
+  id: string;
+  first_name: string;
+  last_name: string;
+  email: string;
+};
+export type CreatePayload = {
```
Comment (Sanmandresy) [inline src/types/user.ts:9]: remove

File: src/types/user.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+export type User = {
```
Comment (Sanmandresy) [inline src/types/user.ts:1]: you're not following the api specification

File: src/providers/user.ts
Comment (Sanmandresy) [inline src/providers/user.ts:1]: you're not handling exceptions here

File: src/providers/region.ts
Comment (Sanmandresy) [inline src/providers/region.ts:1]: same here

File: src/types/bearer.ts
Comment (Sanmandresy) [inline src/types/bearer.ts:1]: change to an object that contains the header instead.

**Review by Sanmandresy — CHANGES_REQUESTED** [review]

File: src/types/user.ts (hunk 3 above)
Comment (Vohizy) [reply src/types/user.ts:9]: why ?

File: src/types/user.ts (hunk 4 above)
Comment (Vohizy) [reply src/types/user.ts:1]: If I have followed the API correctly! and 
Could you please specify which part of the API is incorrect

**Pull request closed without merging**